### PR TITLE
[buffer filter] Remove max_request_time field and code

### PR DESCRIPTION
**Pull request by auni53**
*Description*: This field was deprecated in 1.90 in favour of the new request timeout feature in the http connection manager.
*Risk Level*: Medium-high?
*Testing*: Adjusted config testing and etc. There were never any integration tests for the buffer filter timeout.

### CODE REVIEW

**Comment by auni53** [comment]
Note that I didn't actually remove the field, it's just marked as deprecated. Let me know if I ought to.

File: api/envoy/config/filter/http/buffer/v2/buffer.proto [hunk 1]
```diff
@@ -19,12 +19,7 @@ message Buffer {
   // manager will stop buffering and return a 413 response.
   google.protobuf.UInt32Value max_request_bytes = 1 [(validate.rules).uint32.gt = 0];
 
-  // The maximum number of seconds that the filter will wait for a complete
-  // request before returning a 408 response.
-  // deprecated in favor of http connection manager of :ref:request timeouts
-  // <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.request_timeout>
-  google.protobuf.Duration max_request_time = 2
-      [deprecated = true, (validate.rules).duration = {gt: {}}, (gogoproto.stdduration) = true];
+  google.protobuf.Duration max_request_time = 2 [deprecated = true];
```
Comment (mattklein123) [inline api/envoy/config/filter/http/buffer/v2/buffer.proto]: Hmm yeah, we should have deprecated this in the previous PR. We did mark this as deprecated in release notes for the last release, right? If so, I don't think it matters. But you should remove this field and reserve it.

File: source/extensions/filters/http/buffer/buffer_filter.h [hunk 2]
```diff
@@ -39,12 +39,11 @@ class BufferFilterSettings : public Router::RouteSpecificFilterConfig {
 
   bool disabled() const { return disabled_; }
   uint64_t maxRequestBytes() const { return max_request_bytes_; }
-  std::chrono::seconds maxRequestTime() const { return max_request_time_; }
+  std::chrono::seconds maxRequestTime() const { return std::chrono::seconds(0); }
```
Comment (mattklein123) [inline source/extensions/filters/http/buffer/buffer_filter.h]: Delete function?

File: api/envoy/config/filter/http/buffer/v2/buffer.proto (hunk 1 above)
Comment (auni53) [reply api/envoy/config/filter/http/buffer/v2/buffer.proto]: It was marked as deprecated in the previous PR, I just removed the extra validations. I'll remove the field and reserve it now.

**Pull request merged**